src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+ * @param [in]    included       Boolean variable indicating whether the
+ *                               process/thread participates in the newly created team
+ * @param [out]    new_ucc_team   Pointer to the new team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_create_from_parent(uint64_t my_ep, bool included,
+                                             ucc_team_h parent_team, 
+                                             ucc_team_h *new_ucc_team);
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_get_size routine queries the size of the team. It reflects the
+ * number of unique endpoints in the team.
+ *
+ * @param [in]   ucc_team  Team handle
+ * @param [out]  size      The size of team as number of endpoints
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_size(ucc_team_h ucc_team, uint32_t *size);
+
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_get_my_ep routine queries and returns the endpoint of the
+ * participant invoking the interface.
+ *
+ * @param [out]  ep          Endpoint of the participant calling the routine
+ * @param [in]   ucc_team    Team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_my_ep(ucc_team_h ucc_team, uint64_t *ep);
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_my_ep routine queries and returns all endpoints of all
+ * participants in the team.
+ *
+ * @param [out]     ep          List of endpoints
+ * @param [out]     num_eps     Number of endpoints
+ * @param [in]      ucc_team    Team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_all_eps(ucc_team_h ucc_team, uint64_t **ep, uint64_t
+                                  *num_eps);
+
+
+/**
+ * *************************************************************
+ *                   Collectives Section 
+ * *************************************************************
+ */
+typedef enum {
+    UCC_DT_INT8 = 0,
+    UCC_DT_INT16,
+    UCC_DT_INT32,
+    UCC_DT_INT64,
+    UCC_DT_INT128,
+    UCC_DT_UINT8,
+    UCC_DT_UINT16,
+    UCC_DT_UINT32,
+    UCC_DT_UINT64,
+    UCC_DT_UINT128,
+    UCC_DT_FLOAT16,
+    UCC_DT_FLOAT32,
+    UCC_DT_FLOAT64,
+    UCC_DT_USERDEFINED,
+    UCC_DT_OPAQUE
+} ucc_datatype_t;
+
+typedef struct ucc_coll_buffer_info {
+    uint64_t        mask;
+    void            *src_buffer;
+    ucc_count_t     *src_counts;
+    ucc_aint_t      *src_displacements;
+    void            *dst_buffer;
+    ucc_count_t     *dst_counts;
+    ucc_aint_t      *dst_displacements;
+    ucc_datatype_t  src_datatype;
+    ucc_datatype_t  dst_datatype;
+    uint64_t        flags;
+} ucc_coll_buffer_info_t;
+
+typedef struct ucc_reduction_info {
+    ucc_datatype_t      dt;
+    ucc_reduction_op_t  op;
+    size_t              count;
+} ucc_reduction_info_t;
+
+typedef enum {
+    UCC_ERR_TYPE_LOCAL=0,
+    UCC_ERR_TYPE_GLOBAL=1
+} ucc_error_type_t;
```

Comment:
I'm assuming this field is not future-extendable anyway (no mask) - why not replace it with a boolean?

Reply:
Good point. Though in the first version I don't see something beyond these two values, we can think of other variations. I think for future-proof, we should add a mask.